src/kafka.rs

```diff
@@ -0,0 +1,216 @@
+
+    if std::path::Path::new(&data_folder).exists() && std::fs::read_dir(&data_folder)?.count() > 0 {
+        info!("Alerts already downloaded to {}", data_folder);
+        let count = std::fs::read_dir(&data_folder).unwrap().count();
+        return Ok(count as i64);
+    }
+
+    std::fs::create_dir_all(&data_folder)?;
+
+    info!("Downloading alerts for date {}", date);
+    let url = format!(
+        "https://ztf.uw.edu/alerts/public/ztf_public_{}.tar.gz",
+        date
+    );
+    let output = std::process::Command::new("wget")
+        .arg(&url)
+        .arg("-P")
+        .arg(&data_folder)
+        .output()?;
+    if !output.status.success() {
+        let error_msg = String::from_utf8_lossy(&output.stderr);
+        if error_msg.contains("404 Not Found") {
+            return Err("No alerts found for this date".into());
+        }
+        return Err(format!("Failed to download alerts: {}", error_msg).into());
+    } else {
+        info!("Downloaded alerts to {}", data_folder);
+    }
+
+    let output = std::process::Command::new("tar")
+        .arg("-xzf")
+        .arg(format!("{}/ztf_public_{}.tar.gz", data_folder, date))
+        .arg("-C")
+        .arg(&data_folder)
+        .output()?;
+    if !output.status.success() {
+        return Err("Failed to extract alerts".into());
+    } else {
+        info!("Extracted alerts to {}", data_folder);
+    }
+
+    std::fs::remove_file(format!("{}/ztf_public_{}.tar.gz", data_folder, date))?;
+
+    let count = std::fs::read_dir(&data_folder)?.count();
+
+    Ok(count as i64)
+}
+
+pub async fn consume_alerts(
+    topic: &str,
+    group_id: Option<String>,
+    exit_on_eof: bool,
+    max_in_queue: usize,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let group_id = match group_id {
+        Some(id) => id,
+        None => uuid::Uuid::new_v4().to_string(),
+    };
+
+    let consumer: BaseConsumer = ClientConfig::new()
+        .set("group.id", group_id)
+        .set("bootstrap.servers", "localhost:9092")
+        .set("enable.partition.eof", "false")
+        .set("session.timeout.ms", "6000")
+        .set("enable.auto.commit", "true")
+        .set("auto.offset.reset", "earliest")
+        .create()
+        .unwrap();
+
+    consumer.subscribe(&[topic]).unwrap();
+
+    let metadata = match consumer.fetch_metadata(Some(&topic), std::time::Duration::from_secs(1)) {
+        Ok(m) => m,
+        Err(e) => {
+            return Err(e.into());
+        }
+    };
+
+    let metadata = metadata.topics().iter().next().unwrap();
+
+    let client = redis::Client::open("redis://localhost:6379".to_string()).unwrap();
+    let mut con = client.get_multiplexed_async_connection().await.unwrap();
+    let queue_name = "ZTF_alerts_packets_queue";
+
+    let mut total = 0;
+
+    info!("Reading from kafka topic and pushing to the queue");
+    // start timer
+    let start = std::time::Instant::now();
+    // poll one message at a time
```

Comment:
I feel like there must be a more elegant way to set this up, maybe using channels or something like that. We can explore this at some point.